redsky/streamer.py

```diff
@@ -0,0 +1,57 @@
+from uuid import uuid4
+import os
+from copy import deepcopy as dc
+
+
+def db_store(db, fs_data_name_save_map=None):
+    if fs_data_name_save_map is None:
+        fs_data_name_save_map = {}
+
+    def wrap(f):
+        def wrapped_f(*args, **kwargs):
+            gen = f(*args, **kwargs)
+            for name, doc in gen:
+                fs_doc = dc(doc)
+
+                if name == 'descriptor':
+                    # Mutate the doc here to handle filestore
+                    for data_name in fs_data_name_save_map.keys():
+                        fs_doc['data_keys'][data_name].update(
+                            external='FILESTORE:',
+                            dtype='array')
```

Comment:
He descriptor (whether it is a straight copy of the one that originates from bluesky or an analysis-derived one from a user analysis func) provides the correct dtype. There should be no need to update it here.